src/Test/TestCombatMods.java

```diff
@@ -158,4 +163,84 @@ private boolean testRetribution()
 
     return testPassed;
   }
+
+  private boolean testDamageDealtToIncomeConverter()
+  {
+    boolean testPassed = true; // Assume nothin's busted
+    DamageDealtToIncomeConverter ddtic = DamageDealtToIncomeConverter.instance(testGame, DamageDealtToIncomeConverter.class);
+
+    Unit aa = addUnit(testMap, cinder, UnitModel.SURFACE_TO_AIR, 7, 3);
+    // juice aa to be extra angry and get overkills
+    aa.alterHP(10, true);
+    aa.addUnitModifier(new CODamageModifier(42));
+
+    int currentFundsReturn = 0;
+    // Get lots of free money, and make sure we get the right amount
+    for( int i = 0; i < 100; ++i )
+    {
+      currentFundsReturn += i;
+      ddtic.startTracking(cinder, i);
+      cinder.money = 0;
+
+      // give aa ammo
+      aa.isTurnOver = false;
+      ResupplyEvent event = new ResupplyEvent(null, aa);
+      event.performEvent( testGame.gameMap );
+      GameEventListener.publishEvent(event, testGame);
+
+      // Add the victim, and yeet him
+      addUnit(testMap, venge, UnitModel.TROOP, 7, 4);
+      performGameAction(new BattleLifecycle.BattleAction(testMap, aa, Utils.findShortestPath(aa, 7, 3, testMap), 7, 4), testGame);
+
+      testPassed &= validate(cinder.money == currentFundsReturn * 1000, "    Expected to make "+currentFundsReturn*1000+", but got "+cinder.money+" instead.");
+    }
+    // Remove those amounts, and make sure that works
+    for( int i = 0; i < 55; ++i )
+    {
+      currentFundsReturn -= i;
+      ddtic.stopTracking(cinder, i);
+      ddtic.stopTracking(cinder, i); // Remove twice, just to be sure
+      cinder.money = 0;
+
+      // give aa ammo
+      aa.isTurnOver = false;
+      ResupplyEvent event = new ResupplyEvent(null, aa);
+      event.performEvent( testGame.gameMap );
+      GameEventListener.publishEvent(event, testGame);
+
+      // Add the victim, and yeet him
+      addUnit(testMap, venge, UnitModel.TROOP, 7, 4);
+      performGameAction(new BattleLifecycle.BattleAction(testMap, aa, Utils.findShortestPath(aa, 7, 3, testMap), 7, 4), testGame);
+
+      testPassed &= validate(cinder.money == currentFundsReturn * 1000, "    Expected to make "+currentFundsReturn*1000+", but got "+cinder.money+" instead.");
+    }
+    // Do the second half backwards, for giggles
+    for( int i = 99; i > 42; --i )
```

Comment:
Is this supposed to be a continuation of the previous loop, but reversing the direction? The cutoff isn't even in the same place (though I guess that doesn't matter with how the tests are set up).
This does make the test more complex than it needs to be though. I'm not sure why we even need the second loop (see comment above), but if so you should at least not go out of your way to make the test more difficult to understand.

Reply:
Yes. The intent is to make sure the tracker works regardless of add/remove order, including when there are more removes than adds.
If there was no reason to make the test more complex, I don't think I would have.

Reply:
> If there was no reason to make the test more complex, I don't think I would have.

And yet you overlap the third loop with the second, meaning you execute it 55-43=12 extra times after `currentFundsReturn` is already zero.
I mean, I guess it's worth adding in one order and removing in a different order, but using one loop to add everything and two loops to remove it all in different orders is probably more than needed to prove it does what it says (and is certainly more intricate than is probably needed or expected).

Reply:
The overlap is intentional, though it probably doesn't need to be run 12 times.
It probably also doesn't need to be run every time the game starts, but here we are.

You're probably right about it being more than is needed.
I don't think I'd have bothered writing the test at all if you hadn't suggested it, though - we're already beyond my personal expectations of what's needed/expected, so I covered all my bases.

Reply:
I mean, a confusing-but-correct test is better than no test, but a clear test is yet better. On the other hand, YGWYPF.